File: src/main/java/org/intellij/sdk/lore/LoreFileType.java
package org.intellij.sdk.lore;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class LoreFileType extends LanguageFileType {

    public static final LoreFileType INSTANCE = new LoreFileType();

    private LoreFileType() {
        super(LoreLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Lore File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Lore language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "lore";
    }

    @Override
    public Icon getIcon() {
        return LoreIcons.FILE;
    }

}

File: src/main/kotlin/org/intellij/sdk/lore/LoreFileType.kt
package org.intellij.sdk.lore

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object LoreFileType : LanguageFileType(LoreLanguage.INSTANCE) {
    override fun getName(): String = "Lore File"
    override fun getDescription(): String = "Lore language file"
    override fun getDefaultExtension(): String = "lore"
    override fun getIcon(): Icon = LoreIcons.FILE
}